### PR TITLE
Adjust IProvider2::isAvailable to match IProvider::isAvailable as wel…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…l as IPreview::isAvailable

## Description
IPreview::isAvailable requires FileInfo as argument - IProvider2::isAvailable shall use the same otherwise we are breaking the API.

```
{"reqId":"2SbzswhSScFkJApSWk0H","level":3,"time":"2018-06-20T13:46:41+00:00","remoteAddr":"80.187.104.75","user":"admin","app":"index","method":"GET","url":"/index.php/apps/activity/api/v2/activity/all?format=json&previews=true&since=0","message":"Exception: {\"Exception\":\"TypeError\",\"Message\":\"Argument 1 passed to OC\\Preview\\TXT::isAvailable() must be an instance of OCP\\Files\\File, instance of OC\\Files\\FileInfo given, called in /var/www/owncloud/lib/private/PreviewManager.php on line 221\",\"Code\":0,\"Trace\":\"
#0 /var/www/owncloud/lib/private/PreviewManager.php(221): OC\\Preview\\TXT->isAvailable(Object(OC\\Files\\FileInfo))
#1 /var/www/owncloud/apps/activity/lib/Controller/OCSEndPoint.php(311): OC\\PreviewManager->isAvailable(Object(OC\\Files\\FileInfo))
#2 /var/www/owncloud/apps/activity/lib/Controller/OCSEndPoint.php(238): OCA\\Activity\\Controller\\OCSEndPoint->getPreview('admin', 39081, '/Hacker-Songs/N...')
#3 /var/www/owncloud/apps/activity/lib/Controller/OCSEndPoint.php(181): OCA\\Activity\\Controller\\OCSEndPoint->get(Array)
#4 /var/www/owncloud/apps/activity/lib/Controller/EndPoint.php(68): OCA\\Activity\\Controller\\OCSEndPoint->getFilter(Array)
#5 [internal function]: OCA\\Activity\\Controller\\EndPoint->getFilter('all')
#6 /var/www/owncloud/lib/private/AppFramework/Http/Dispatcher.php(159): call_user_func_array(Array, Array)
#7 /var/www/owncloud/lib/private/AppFramework/Http/Dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Activity\\Controller\\EndPoint), 'getFilter')
#8 /var/www/owncloud/lib/private/AppFramework/App.php(103): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Activity\\Controller\\EndPoint), 'getFilter')
#9 /var/www/owncloud/lib/private/AppFramework/Routing/RouteActionHandler.php(46): OC\\AppFramework\\App::main('EndPointControl...', 'getFilter', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)
#10 [internal function]: OC\\AppFramework\\Routing\\RouteActionHandler->__invoke(Array)
#11 /var/www/owncloud/lib/private/Route/Router.php(342): call_user_func(Object(OC\\AppFramework\\Routing\\RouteActionHandler), Array)
#12 /var/www/owncloud/lib/base.php(919): OC\\Route\\Router->match('/apps/activity/...')
#13 /var/www/owncloud/index.php(55): OC::handleRequest()
#14 {main}\",\"File\":\"/var/www/owncloud/lib/private/Preview/TXT.php\",\"Line\":97}"}
```

## Motivation and Context
Don't break our public api

## How Has This Been Tested?
- retest activity app - @michaelstingl has detailed steps

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/private/Preview/Image.php [hunk 1]
```diff
@@ -59,7 +60,7 @@ public function getThumbnail(File $file, $maxX, $maxY, $scalingUp) {
 	/**
 	 * @inheritdoc
 	 */
-	public function isAvailable(File $file) {
+	public function isAvailable(FileInfo $file) {
```
Comment (PVince81) [inline lib/private/Preview/Image.php]: remove the `File` import everywhere ?

**Review by PVince81 — CHANGES_REQUESTED** [review]
approach looks good, please adjust the imports

**Comment by michaelstingl** [comment]
> @michaelstingl has detailed steps

Sorry, no detailed steps. Just broken on my private server. But I can test the PR as soon a patch for stable10 is available.

**Comment by PVince81** [comment]
@DeepDiver1975 probably also needs some test adjustments and use a FileInfo mock instead of File ?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 probably also needs some test adjustments and use a FileInfo mock instead of File ?

adjusted

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@DeepDiver1975 please backport

**Comment by michaelstingl** [comment]
I can confirm, fixed with the `owncloud-10.0.9RC2.tar.bz2`

**Comment by phil-davis** [comment]
Backport ``stable10`` #31862

**Pull request merged**